osu.Game.Tests/Visual/Multiplayer/TestSceneMultiplayerGameplayLeaderboard.cs

```diff
@@ -70,14 +52,11 @@ public override void SetUpSteps()
 
                 var playable = Beatmap.Value.GetPlayableBeatmap(Ruleset.Value);
 
-                for (int i = 0; i < users; i++)
-                    spectatorClient.StartPlay(i, Beatmap.Value.BeatmapInfo.OnlineBeatmapID ?? 0);
+                foreach (var user in users)
+                    SpectatorClient.StartPlay(user, Beatmap.Value.BeatmapInfo.OnlineBeatmapID ?? 0);
 
-                spectatorClient.Schedule(() =>
-                {
-                    Client.CurrentMatchPlayingUserIds.Clear();
-                    Client.CurrentMatchPlayingUserIds.AddRange(spectatorClient.PlayingUsers);
-                });
+                // Todo: This is REALLY bad.
+                Client.CurrentMatchPlayingUserIds.AddRange(users);
```

Comment:
What's this referring to? The fact that that is mutable? It looks pretty innocuous on a loose read-through...

Reply:
The fact that it's mutable, yes. It's something I've addressed in the past so it's in my sights, but I thought it was the cause of all other other test failures at the time.